errors.go

```diff
@@ -0,0 +1,84 @@
+package rpc
+
+import "errors"
+
+// ResponseErr is an enum type for providing error type
+// information over the wire between rpc server and client.
+type ResponseErr int
+
+const (
+	// NonRPCErr is an error that hasn't arisen from the gorpc package.
+	NonRPCErr ResponseErr = iota
+	// ServerErr is an error that has arisen on the server side.
+	ServerErr
+	// ClientErr is an error that has arisen on the client side.
+	ClientErr
+)
+
+// ServerError indicates that error originated in server
+// specific code.
+type ServerError struct {
+	msg string
+}
+
+func (s *ServerError) Error() string {
+	return s.msg
+}
+
+// NewServerError wraps an error in the ServerError type.
+func NewServerError(err error) error {
+	return &ServerError{err.Error()}
+}
+
+// ClientError indicates that error originated in client
+// specific code.
+type ClientError struct {
+	msg string
+}
+
+func (c *ClientError) Error() string {
```

Comment:
Q: How come golint does not complain here?

Reply:
I think because it is **the** `error` interface, it is somewhat redundant to comment it. But I am just guessing.